diff --git a/mdbx_load.c b/mdbx_load.c
index 8c9f20b..4047ded 100644
--- a/mdbx_load.c
+++ b/mdbx_load.c
@@ -4080,7 +4080,7 @@ int main(int argc, char *argv[]) {
         goto bailout;
       }
 
-      if (batch == 10000 || txn_info.txn_space_dirty > MEGABYTE * 256) {
+      if (txn_info.txn_space_dirty > MEGABYTE * 1024 * 2) {
         err = mdbx_txn_commit(txn);
         if (unlikely(err != MDBX_SUCCESS)) {
           error("mdbx_txn_commit", err);
